### PR TITLE
Daniel/test sets

### PR DESCRIPTION
**Pull request by daniel020403**
## Description
Added tests for Sets and GSets.

### CODE REVIEW

**Review by pjaclark — CHANGES_REQUESTED** [review]
testGSet needs to try and remove an item and fail.

After checking it updated, do the same as testSet by removing user2, and having that fail as we'd expect.

**Comment by daniel020403** [comment]
> testGSet needs to try and remove an item and fail.
> 
> After checking it updated, do the same as testSet by removing user2, and having that fail as we'd expect.

Exception test for when removing an item from a GSet added.
@pjaclark

File: src/test/java/com/basho/riak/client/api/commands/itest/ITestDatatype.java [hunk 1]
```diff
@@ -192,37 +190,92 @@ public void testNotFoundHyperLogLog() throws ExecutionException, InterruptedExce
...
+        assertTrue(updatedSet.contains(BinaryValue.create("user2")));
+        assertTrue(updatedSet.contains(BinaryValue.create("user3")));
+        assertFalse(updatedSet.contains(BinaryValue.create("user4")));
+
+        final FetchSet.Response removeItemResponse = client.execute(fetchSet);
+        Context ctx = removeItemResponse.getContext();
+        try {
+            SetUpdate suRemoveItem = new SetUpdate().remove("user2");
+            UpdateSet updateRemove = new UpdateSet.Builder(location, suRemoveItem).withContext(ctx).build();
+            client.execute(updateRemove);
```
Comment (pjaclark) [inline src/test/java/com/basho/riak/client/api/commands/itest/ITestDatatype.java:268]: After this, there should be an Assert that always fails as it should never get to the assert line due to an exception being thrown.

File: src/test/java/com/basho/riak/client/api/commands/itest/ITestDatatype.java [hunk 2]
```diff
@@ -192,37 +190,92 @@ public void testNotFoundHyperLogLog() throws ExecutionException, InterruptedExce
...
+        } catch (Exception e) {
```
Comment (pjaclark) [inline src/test/java/com/basho/riak/client/api/commands/itest/ITestDatatype.java]: Change to IllegalArgumentException and let others through.

**Comment by pjaclark** [comment]
LGTM

**Review by pjaclark — APPROVED** [review]

**Review by Bob-The-Marauder — APPROVED** [review]
Java is not my strong suit but everything looks ok to me as far as I can see.

**Pull request merged**